learning_wavelets/models/exact_recon_unet.py

```diff
@@ -0,0 +1,169 @@
+import tensorflow as tf
+import numpy as np
+from tensorflow.keras.layers import Layer, Conv2D, LeakyReLU, PReLU, UpSampling2D, MaxPooling2D, Activation
+from tensorflow.keras.models import Model
+from math import log, ceil, floor
```

Comment:
For the padding/cropping, use `tensorflow` functions to work on tensors.

I think this is why you needed to use the eager mode.